client/image_push_test.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+func TestImagePushServerError(t *testing.T) {
+	client := &APIClient{
+		HTTPCli: newMockClient(errorMockResponse(http.StatusInternalServerError, "Server error")),
+	}
+	_, err := client.ImagePush(context.Background(), "image", "auth")
+	if err == nil || !strings.Contains(err.Error(), "Server error") {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestImagePushWrongError(t *testing.T) {
```

Comment:
I think this case is duplicated with `TestImagePushServerError`. It doesn't help. Could we consider to remove this? WDYT?